db/db.go

```diff
@@ -0,0 +1,98 @@
+package db
+
+import (
+	"encoding/json"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/nosql"
+)
+
+var revokedCertsTable = []byte("revoked_x509_certs")
+
+// Config represents the JSON attributes used for configuring a step-ca DB.
+type Config struct {
+	Type string `json:"type"`
+	Path string `json:"path"`
+}
+
+// DB is a wrapper over the nosql.DB interface.
+type DB struct {
+	nosql.DB
+}
+
+// New returns a new database client that implements the nosql.DB interface.
+func New(c *Config) (*DB, error) {
+	switch strings.ToLower(c.Type) {
+	case "bbolt":
+		db := &nosql.BoltDB{}
+		err := db.Open(c.Path)
+		return &DB{db}, err
+	default:
+		return nil, errors.Errorf("unsupported db.type '%s'", c.Type)
+	}
+}
+
+// Init database with all necessary tables.
+func (db *DB) Init(c *Config) (*DB, error) {
+	var err error
+	if db == nil {
+		if db, err = New(c); err != nil {
+			return nil, err
+		}
+	}
+	tables := [][]byte{revokedCertsTable}
+	for _, b := range tables {
+		if err = db.CreateTable(b); err != nil {
+			return nil, errors.Wrapf(err, "error creating table %s",
+				string(b))
+		}
+	}
+	return db, nil
+}
+
+// RevokedCertificateInfo contains information regarding the certificate
+// revocation action.
+type RevokedCertificateInfo struct {
+	Serial        string
+	ProvisionerID string
+	Reason        int
+	RevokedAt     time.Time
+}
```

Comment:
Store the ReasonCode.

Reply:
this is already the reason code. So are you asking for it to be stored as `ReasonCode` instead of `Reason`?

Reply:
I thought it was the string